staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -476,7 +476,18 @@ type DeploymentStatus struct {
 	// +optional
 	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty" protobuf:"varint,5,opt,name=unavailableReplicas"`
 
-	// Represents the latest available observations of a deployment's current state.
+	// Represents the latest observations of a deployment's current state (described by the observedGeneration).
+	// Deployments are given the following conditions automatically:
+	// Available is true when the number of available replicas under the deployment's replica sets is
+	// greater than the user's defined minimum availability. The minimum availability is controlled by the
+	// update strategy - the Recreate strategy has a minimum availability of zero (always Available), and
+	// the Rolling strategy defaults to a minimum availability of zero unless replicas is greater than 1,
+	// the maxSurge parameter is specified, or maxUnavailable is smaller than the number of replicas.
```

Comment:
I think it is better to describe the general working of the condition before defining special cases

Reply:
Agree

Reply:
Added a sentence above, and tried to capture that "available condition is a lie and can't be relied on in this case".